### PR TITLE
cpu/gd32v: add periph_gpio_ll and periph_gpio_ll_irq support

### PR DESCRIPTION
**Pull request by gschorcht**
### Contribution description

This PR provides the `periph_gpio_ll` and `periph_gpio_ll_irq` support for GD32VF103. Level triggered interrupts are emulated.

`periph_gpio_ll_irq` could be split off from this PR as a separate PR if necessary.

### Testing procedure

Use any GD32V board and connect PA0 -> PB0 and PA1 -> PB1 where PA is the output port and PB the input port. With these connections `tests/periph_gpio_ll` should work.
```
BOARD=sipeed-longan-nano make -j8 -C tests/periph_gpio_ll flash term
```

If necessary, change the input and output pins by setting the environment variables and connect the corresponding pins, for example for `seeedstudio-gd32` PA1 -> PB8 and PA8 -> PB9:
```
PIN_OUT_0=1 PIN_OUT_1=8  PIN_IN_0=8 PIN_IN_1=9 BOARD=seedstudio-gd32 make -j8 -C tests/periph_gpio_ll flash term
```

### CODE REVIEW

File: cpu/gd32v/periph/gpio_ll_irq.c [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+            break;
+        case GPIO_TRIGGER_LEVEL_HIGH:
+            EXTI->RTEN |= pin_mask;
+            _h_level_triggered |= pin_mask;
+            break;
+        case GPIO_TRIGGER_LEVEL_LOW:
+            EXTI->FTEN |= pin_mask;
+            _l_level_triggered |= pin_mask;
+            break;
+    }
```
Comment (maribu) [inline cpu/gd32v/periph/gpio_ll_irq.c:123]:
```suggestion
    switch (trig) {
    case GPIO_TRIGGER_EDGE_RISING:
        EXTI->RTEN |= pin_mask;
        break;
    case GPIO_TRIGGER_EDGE_FALLING:
        EXTI->FTEN |= pin_mask;
        break;
    case GPIO_TRIGGER_EDGE_BOTH:
        EXTI->RTEN |= pin_mask;
        EXTI->FTEN |= pin_mask;
        break;
    case GPIO_TRIGGER_LEVEL_HIGH:
        EXTI->RTEN |= pin_mask;
        _h_level_triggered |= pin_mask;
        break;
    case GPIO_TRIGGER_LEVEL_LOW:
        EXTI->FTEN |= pin_mask;
        _l_level_triggered |= pin_mask;
        break;
    }
```

**Review by maribu — APPROVED** [review]
Thx :) Code looks good and the app should catch any bugs. Please squash right away.

File: cpu/gd32v/periph/gpio_ll_irq.c (hunk 1 above)
Comment (gschorcht) [reply cpu/gd32v/periph/gpio_ll_irq.c:123]: When I wrote this switch case construct, I knew you were going to ask for a change in style 😉 I still don't find it that readable. I will change andsquah it directly.

**Comment by maribu** [comment]
bors merge

**Comment by maribu** [comment]
manual merge train packing :steam_locomotive: :train: :train:

bors retry

**Comment by maribu** [comment]
bors cancel
bors retry

Come on, let's get two PRs in the merge train :)

**Comment by gschorcht** [comment]
> Build failed (retrying...):

Cool, Rust catched a problem in `#endif DOXYGEN` that the preprocessor was accpeting.

**Comment by gschorcht** [comment]
bors merge

**Comment by benpicco** [comment]
bors merge

**Comment by gschorcht** [comment]
Thanks for reviewing.

**Pull request merged**